src/benchmarks/micro/corefx/System.Buffers/ReadOnlySequenceTests.cs

```diff
@@ -28,27 +29,73 @@ public class ReadOnlySequenceTests<T>
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+            consume += sequence.FirstSpan.Length; consume += sequence.FirstSpan.Length;
+
+            return consume;
+        }
+
+        [GlobalSetup(Targets = new [] { nameof(FirstSpanSingleSegment) })]
+        public void SetupFirstSpanSingleSegment() => SetupSingleSegment();
+
+        [GlobalSetup(Targets = new [] { nameof(FirstSpanMemory) })]
+        public void MemoryFirstSpanMemory() => MemorySegment();
+
+        [GlobalSetup(Targets = new [] { nameof(FirstSpanTenSegments) })]
+        public void SetupFirstSpanTenSegments() => SetupTenSegments();
+#endif
+
         [Benchmark(OperationsPerInvoke = 10)]
         public long SliceArray() => Slice(new ReadOnlySequence<T>(_array));
 
         [Benchmark]
-        public int IterateTryGetMemory() => IterateTryGet(new ReadOnlySequence<T>(new ReadOnlyMemory<T>(_array)));
+        public int IterateTryGetMemory() => IterateTryGet(new ReadOnlySequence<T>(_memory));
 
         [Benchmark]
-        public int IterateForEachMemory() => IterateForEach(new ReadOnlySequence<T>(new ReadOnlyMemory<T>(_array)));
+        public int IterateForEachMemory() => IterateForEach(new ReadOnlySequence<T>(_memory));
 
         [Benchmark]
-        public int IterateGetPositionMemory() => IterateGetPosition(new ReadOnlySequence<T>(new ReadOnlyMemory<T>(_array)));
+        public int IterateGetPositionMemory() => IterateGetPosition(new ReadOnlySequence<T>(_memory));
 
         [Benchmark(OperationsPerInvoke = 16)]
-        public int FirstMemory() => First(new ReadOnlySequence<T>(new ReadOnlyMemory<T>(_array)));
+        public int FirstMemory() => First(new ReadOnlySequence<T>(_memory));
 
         [Benchmark(OperationsPerInvoke = 10)]
-        public long SliceMemory() => Slice(new ReadOnlySequence<T>(new ReadOnlyMemory<T>(_array)));
+        public long SliceMemory() => Slice(new ReadOnlySequence<T>(_memory));
 
         [GlobalSetup(Targets = new [] { nameof(IterateTryGetSingleSegment), nameof(IterateForEachSingleSegment), nameof(IterateGetPositionSingleSegment), nameof(FirstSingleSegment), nameof(SliceSingleSegment) })]
         public void SetupSingleSegment() => _startSegment = _endSegment = new BufferSegment<T>(new ReadOnlyMemory<T>(_array));
 
+        [GlobalSetup(Targets = new [] { nameof(FirstMemory), nameof(SliceMemory), nameof(IterateTryGetMemory), nameof(IterateForEachMemory), nameof(IterateGetPositionMemory) })]
```

Comment:
Don't we need this setup for `FirstSpanMemory` as well which is if/def'd for .NET Core 3.0 only?

Otherwise, what's the value of `_memory` there?
https://github.com/dotnet/performance/pull/410/files#diff-19d9aaa505dcc18077d456ddc9a86808R37

We might need to if/def this setup too, if that's the suggested approach for adding benchmarks specific to .NET Core 3.0 APIs.

Reply:
There is GlobalSetup for FirstSpan* benchmarks inside the ifdef which will call functions outside of ifdef instantiating the _memory

Reply:
Oh right, missed that one. That works :)